test/shared/roundtrip.e2e.test.ts

```diff
@@ -0,0 +1,56 @@
+import { test, expect, Page } from "@playwright/test";
+import {
+    switchMode,
+    clearEditor,
+    editorSelector,
+    enterTextAsMarkdown,
+} from "../e2e-helpers";
+
+test.describe.serial("roundtrip tests", () => {
+    let page: Page;
+    test.beforeAll(async ({ browser }) => {
+        page = await browser.newPage();
+        await page.goto("/empty.html");
+        await switchMode(page, "markdown");
+    });
+
+    test.afterAll(async () => {
+        await page.close();
+    });
+
+    for (const [markdown] of [
+        //Basic commonmark
+        ["plain"],
+        ["*italic*"],
+        ["_italic_"],
+        ["**bold**"],
+        ["__bold__"],
+        ["# H1"],
+        ["## H2"],
+        ["[link](http://www.example.com)"],
+        ["![Image](http://www.example.com/pretty.png)"],
+        ["> blockquote"],
+        ["* List Item"],
+        ["- List Item"],
+        ["1. List Item"],
+        ["2) List Item"],
+        ["lol\n\n---\n\nlmao"],
+        ["lol\n\n***\n\nlmao"],
+        ["`code`"],
+        //TODO: Codeblock does weird things roundtripping: Adds an extra space
+        //['```javascript\ncodeblock\n```' ],
```

Comment:
👍 Good knowledge for a potential follow up ticket (in case it is worth fixing of course).